### PR TITLE
Transition to pip package with pyproject.toml

### PR DESCRIPTION
**Pull request by patryk-kowalski-ds**
PR Type:
**Refactoring**

___
PR Description:
**This PR transitions the project from using a requirements.txt file for dependency management to using a pyproject.toml file. This change allows for more robust package management and makes the project more compatible with modern Python packaging tools. The PR also updates the Dockerfiles to reflect this change.**

___
PR Main Files Walkthrough:
-`setup.py`: Added a setup function for compatibility with legacy tools.
-`Dockerfile.github_action`: Replaced the addition and installation of requirements.txt with pyproject.toml.
-`docker/Dockerfile`: Similar changes as in Dockerfile.github_action.
-`docker/Dockerfile.lambda`: Replaced the addition and installation of requirements.txt with pyproject.toml and updated the version of mangum.
-`pyproject.toml`: Added a comprehensive project configuration including build system, project details, dependencies, and scripts.
-`requirements.txt`: Removed all specific dependencies and replaced with a reference to the local directory.

### CODE REVIEW

**Comment by patryk-kowalski-ds** [comment]
The authors/maintainers should be filled in, we left it empty

**Comment by mrT23** [comment]
Preparing changelog updates...

**Comment by mrT23** [comment]
**Changelog updates:**

## 2023-07-27

### Changed
- Transitioned the project from using a requirements.txt file for managing dependencies to using a pyproject.toml file, making the project a pip package for better dependency management.
- Updated Dockerfiles to install dependencies from pyproject.toml instead of requirements.txt.
- Updated the version of mangum in Dockerfile.lambda.


>to commit the new content to the CHANGELOG.md file, please type:
>'/update_changelog -commit'

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# for compatibility with legacy tools
+# see: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html
+from setuptools import setup
+
+setup()
```
Comment (okotek) [inline setup.py:5]:
```suggestion
setup()

```

**Review by okotek — APPROVED** [review]

**Comment by okotek** [comment]
Thank you for your contribution @patryk-kowalski-ds

**Pull request merged**